src/shogun/machine/gp/GaussianLikelihood.cpp

```diff
@@ -121,9 +119,9 @@ SGVector<float64_t> CGaussianLikelihood::get_log_probability_derivative_f(
 
 	// compute derivatives of log probability wrt f
 	if (i == 1)
-		eigen_result/=CMath::sq(m_sigma);
+		eigen_result/=CMath::exp(m_log_sigma*2.0);
 	else if (i == 2)
-		eigen_result=-VectorXd::Ones(result.vlen)/CMath::sq(m_sigma);
+		eigen_result=-VectorXd::Ones(result.vlen)/CMath::exp(m_log_sigma*2.0);
```

Comment:
I quite liked your get_sigma() get_width trick above as it makes the code look nicer.

Any reasons why not ?